### ISSUE TITLE
Request: Configurable path to config.yaml?

### ISSUE THREAD

**Issue by beporter** [puphpet/puphpet#1429]
There are some situations where it would be quite helpful to be able to tell the Vagrantfile and the puppet manifests to load puphpet configurations from a different file path (other than the hard-coded `puphpet/config.yaml`.) One example of this is #1320.

I think this could be accomplished by defining the path to `config.yaml` once at the top of the Vagrantfile (as is already the case), but then passing that path to the puppet provisioner using facter. All of the `*.pp` manifests could use the facter-supplied path for importing the proper sub-section of configs instead of having it hard-coded into the top of every single one of them like it is currently.

Example from current [Apache.pp](https://github.com/puphpet/puphpet/blob/0d58fcf3aace33c6efd4b254e29575d10af7c6ab/archive/puphpet/puppet/nodes/Apache.pp#L1):
```puppet
if $yaml_values == undef { $yaml_values = loadyaml('/vagrant/puphpet/config.yaml') }
```

In other words, [all of these matches](https://github.com/puphpet/puphpet/search?q="config.yaml"&type=Code&utf8=✓) (excluding documentation) should really be a single variable definition.

Normal usage of the entire package wouldn't be effected by this change at all, but it would consolidate the path into a single place, making it realistic to move the file (not to mention making the code more DRY.)

**Comment by beporter** [comment]
Another alternative is to move `config.yaml` and `files/` out of the `puphpet/` folder. Both of these are *intended* to be user-changeable. But everything else in the `puphpet/` folder is completely static-- those files are just straight copies of the relevant repos supporting this project.

I believe the "serviceable" files should live in a separate place from the static assets. This folder structure, for example, would make things a lot easier (for me):

```
├── puphpet/               # Static "vendor" assets. No reason to ever play in here.
│   ├── puppet/
│   ├── ruby/
│   └── shell/
├── puphpet-config/        # Optional user-serviceable configs and scripts.
│   │                      #   `vagrant up` should work even if this folder is missing.
│   ├── dot/
│   ├── exec-always/
│   ├── exec-once/
│   ├── startup-always/
│   ├── startup-once/
│   ├── config-local.yaml  # Optional untracked "local" overrides config.
│   │                      #   (Currently named `config-custom.yaml`.)
│   └── config.yaml        # Primary config.
└── Vagrantfile            # Should really be in `puphpet/` but we can't do anything about this.
```

This setup mirrors how a lot of other tools work:

* Git expects `/.git/`, `/.gitignore`, `/.gitmodules` and `.gitattributes` in the project root.
* Composer expects `composer.json` and `composer.lock` in the root.
* NPM expects `package.json` in the root.
* PHPUnit expects `phpunit.xml` and `phpunit.xml.dist` in the root.
* Many tools will use an `.editorconfig` in the root.
* Vagrant expects its `Vagrantfile` in the root.
* Grunt expects a `Gruntfile.js` in the root.

Granted, controlling the path to some of these files is actually possible with some tools, but almost all of them either _expect their config file to live in the project root_, or let you define where to find it. Of those examples, I think only Vagrant doesn't allow you to do **either**. ...And Puphpet, of course. :sweat_smile:

The reasoning for splitting things up like this and making configs optional is that we could get the contents of the `puphpet/` folder from a tool like Composer a lot more easily. The Vagrantfile is still the entry point for all things Puphpet-- it should be possible to load manifests and run provisioners from `vendor/puphpet/puphpet/puppet/` instead of `puphpet/puppet/`. Then all of the stuff a user might ever need to change for their specific project is isolated to the `puphpet-config/` folder.
